ingredient/ingredient-event-hub-namespace/src/functions.ts

```diff
@@ -8,5 +8,12 @@ export class EventHubNamespaceUtils extends BaseUtility {
         const profile = util.create_resource_name("ehn", null, true);
         return profile;
     } 
+
+    public async get_resource_profile(): Promise<string> {
```

Comment:
This sounds like something we should move to corutils?

Reply:
I would actually argue that specific naming utilities should rest with the ingredient. Having to include CoreUtils with a resource naming convention, that would be a mess. You should be able to call and ingredient, and give it a name, the function should concatenate the name, *and* use a guard style function to verify it matches what azure will accept. Length, case, characters, etc. Easily accomplishable via `RegEx`.

Reply:
That's correct for specific naming. This function looked more generic that returns a rg/name pair. Though looking at it further .create_resource_name() isn't valid. Need to pass at least a type + name params.

Reply:
Updated PR.
Sorry guys... naming is still making my head spin a bit.  There appears to be a couple different patterns currently used in Bake around naming within either the specific ingredient or core utils.  I think it would help to see a some sample recipes that cover various scenarios.  Here are a few ideas for scenarios - 1) Recipes has an ingredient that needs to reference another ingredient that was created within the same recipe  2) Recipe has an ingredient that was created in a different recipe such as a shared resource  3) Recipe has an ingredient that creates a centralized resource (not region specific)

Reply:
Can you open an issue on that to track?

Reply:
Yes no problem.  I added https://github.com/HomecareHomebase/azure-bake/issues/59.

Reply:
modified to accept params